src/NDItk/multigroup/Metadata/test/Metadata.test.cpp

```diff
@@ -0,0 +1,136 @@
+// include Catch2
+#include <catch2/catch_test_macros.hpp>
+#include <catch2/matchers/catch_matchers_floating_point.hpp>
+using Catch::Matchers::WithinRel;
+
+// what we are testing
+#include "NDItk/multigroup/Metadata.hpp"
+
+// other includes
+
+// convenience typedefs
+using namespace njoy::NDItk;
+using Metadata = multigroup::Metadata;
+
+std::string chunk();
+void verifyChunk( const Metadata& );
+
+SCENARIO( "Metadata" ) {
+
+  GIVEN( "valid data for a Metadata instance" ) {
+
+    std::string record = chunk();
+
+    WHEN( "the data is given explicitly" ) {
+
+      std::string zaid = "92235.711nm";
+      std::string name = "mendf71x";
+      std::string source = "12/22/2011";
+      std::string process = "08/07/2013";
+      double awr = 233.0248;
+      double weight = 235.043937521619;
+      double temperature = 2.53e-8;
+      double dilution = 1e+10;
+      unsigned int groups = 618;
+      unsigned int upscatter = 0;
+      unsigned int downscatter = 617;
+      unsigned int order = 5;
+      unsigned int reactions = 7;
+      Metadata chunk( std::move( zaid ), std::move( name ),
+                      std::move( source ), std::move( process ),
+                      awr, weight, temperature, dilution,
+                      groups, reactions );
+
+      THEN( "a Metadata can be constructed and members can "
+            "be tested" ) {
+
+        verifyChunk( chunk );
+      } // THEN
+
+      THEN( "the record can be printed" ) {
+
+        std::string buffer;
+        auto output = std::back_inserter( buffer );
+        chunk.print( output );
+
+        CHECK( buffer == record );
+      } // THEN
+    } // WHEN
+
+    WHEN( "the data is read using iterators" ) {
+
+      auto iter = record.begin();
+      auto end = record.end();
+
+      auto readKey = [] ( auto&& iter, auto&& end ) {
```

Comment:
I don't follow the "auto readKey = []" part.

Reply:
Mostly the [] part.

Reply:
This is a lambda. It allows us to define a function in the code. the ```[]``` is what we refer to as the capture for the lambda (i.e. things the lanbda needs to do its job but that you cannot give in the lambda's arguments). In this case, ```[]``` means that nothing needs to be captured.

For example:
```
int id = 4;
auto compare = [id] ( auto&& value ) { return value == id; };
```
In this case the value of ```id``` gets captured by value in the lambda.